src/main/java/com/nylas/AccountQuery.java

```diff
@@ -1,5 +1,96 @@
 package com.nylas;
 
+import okhttp3.HttpUrl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 public class AccountQuery extends RestfulQuery<AccountQuery> {
 
+	private String metadataSearch;
+	private List<String> metadataKeys;
+	private List<String> metadataValues;
+	private List<String> metadataPairs;
```

Comment:
There seems to be quite a lot of repeated code between the different query classes now.  I think it's worth looking for a good way to reduce it.  What do you think about having a MetadataQuery (or similar name) object that can be passed to, and included in, each of these queries, rather than each query having it's own separate metadata fields and methods?  The existing EventQuery methods could be deprecated.